pkg/utils/listener/listener.go

```diff
@@ -0,0 +1,45 @@
+package listener
+
+import (
+	"context"
+	"net"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+// GRPCListener handles gRPC connections using a bufconn listener. This is useful for
+// testing when it's unnecessary to have a live gRPC server running. The normal
+// workflow is to call New() to start the listener, Connect() to start a gRPC
+// connection under which to send client calls, Close() to close the connection, and
+// Release() to close the listener.
+type GRPCListener struct {
```

Comment:
Nice - I like how you've compiled all this functionality into a single place to share between trtl and gds! It simplifies things which is nice! 

Would you mind if we put this in `pkg/utils/bufconn/listener.go`? - listener is a bigger thing and I think we'll remember bufconn for grpc tests a bit easier.

Reply:
Not a problem with me, I was trying to avoid having conflicting names with other packages to keep things simple, but I don't have a better suggestion either.